### ISSUE TITLE
Task maintenance:set:if-needed always sets the store in maintenance mode

### ISSUE THREAD

**Issue by VivekShingala** [jalogut/magento2-deployer-plus#65]
Hi @jalogut 

While deploying the release on server, the task maintenance:set:if-needed always set the site into maintenance mode. Hence it's not zero downtime deployment.

Executing task maintenance:set:if-needed
✔ Executing task maintenance:set
✔ Ok

**Comment by sprankhub** [comment]
As you can see here:

https://github.com/jalogut/magento2-deployer-plus/blob/267c2afaa8f7a012ce4c08c35a9f64aa58ed4f12/recipe/magento_2_2/maintenance.php#L14

the maintenance mode will be set if `database_upgrade_needed` or `config_import_needed` is set.

`database_upgrade_needed` checks `bin/magento setup:db:status`:

https://github.com/jalogut/magento2-deployer-plus/blob/267c2afaa8f7a012ce4c08c35a9f64aa58ed4f12/recipe/magento_2_2/database.php#L18-L33

`config_import_needed` checks `bin/magento app:config:status`:

https://github.com/jalogut/magento2-deployer-plus/blob/267c2afaa8f7a012ce4c08c35a9f64aa58ed4f12/recipe/magento_2_2_5/config.php#L18-L33

So either your database needs to be upgraded every time or config needs to be imported every time.

For us, the maintenance mode is only set when necessary, so this seems to be working in general.

**Comment by VivekShingala** [comment]
Hi @sprankhub 

Yes, I already checked the code behind the task `maintenance:set:if-needed` and I was still facing the issue. Upon checking further, I found that there is some issue with Magento and we need to update this parameter in my.cnf file.

In order for MySQL `TIMESTAMP` fields to follow the preferences and composition expected by Magento's declarative schema architecture, the system variable `explicit_defaults_for_timestamp` must be set to `on`.

If this setting is not enabled, `setup:db:status` will always report that `Declarative Schema is not up to date`

More info here: [Added note regarding use of explicit_defaults_for_timestamps=on](https://github.com/davidalger/devdocs/commit/432e3be8d3ed210f742fe3c40f0c29044cbf96b4)
